src/server/worker.cc

```diff
@@ -442,115 +441,144 @@ Status Worker::EnableWriteEvent(int fd) {
 }
 
 Status Worker::Reply(int fd, const std::string &reply) {
-  std::unique_lock<std::mutex> lock(conns_mu_);
-  auto iter = conns_.find(fd);
-  if (iter != conns_.end()) {
-    iter->second->SetLastInteraction();
-    redis::Reply(iter->second->Output(), reply);
+  if (ConnMap::accessor accessor; conns_.find(accessor, fd)) {
+    accessor->second->SetLastInteraction();
+    redis::Reply(accessor->second->Output(), reply);
     return Status::OK();
   }
 
   return {Status::NotOK, "connection doesn't exist"};
 }
 
 void Worker::BecomeMonitorConn(redis::Connection *conn) {
```

Comment:
Hi @mapleFU ,

Sorry for delay reply.
I'd like to use this way on all accessors to mitigate potential inconsistency in `monitor_conns_` and  `conns_`.

```suggestion
void Worker::BecomeMonitorConn(redis::Connection *conn) {
  ConnMap::accessor accessor;
  ConnMap::accessor monitor_accessor;
  bool find_conn = conns_.find(accessor, conn->GetFD());
  bool find_monitor = monitor_conns_.find(monitor_accessor, conn->GetFD());
  if (find_conn) {
    conns_.erase(accessor);
  }
  if (find_monitor) {
    monitor_accessor->second = conn;
  } else {
    monitor_conns_.insert(monitor_accessor, std::make_pair(conn->GetFD(), conn));
  }
  srv->IncrMonitorClientNum();
  conn->EnableFlag(redis::Connection::kMonitor);
}
```

Best Regards,
Edward